baseband/vlbi_base/base.py

```diff
@@ -129,15 +129,39 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
                  sample_rate=None):
 
         if frames_per_second is None and sample_rate is None:
-            frames_per_second = self._get_frame_rate(fh_raw, type(header0))
+            try:
+                frames_per_second = self._get_frame_rate(fh_raw,
+                                                         type(header0))
+            except Exception as exc:
+                exc.args += ("_get_frame_rate raised {e.__class__.__name__} "
+                             "error before reading one second of data.  File "
+                             "may have less than one second of data, in "
+                             "which case framerate must manually be passed "
+                             "to file opener.  File could also be "
+                             "corrupt.".format(e=exc),)
+                raise
 
         super(VLBIStreamReaderBase, self).__init__(
             fh_raw, header0, nchan, bps, complex_data, thread_ids,
             samples_per_frame, frames_per_second, sample_rate)
 
     @staticmethod
     def _get_frame_rate(fh, header_class):
-        """Returns the number of frames in one second of data."""
+        """
+        Returns the number of frames in one second of data.
+
+        ``_get_frame_rate`` is called when the number of frames
```

Comment:
I would just describe what the function does, not where it is used (or maybe mention this at the very end).